### PR TITLE
Remove PATH assumption from fhs-userenv.

### PR DESCRIPTION
**Pull request by kevincox**
Previously is was assumed that bash was in the path when calling the
environment setup script. This changes all of the references of bash to
be absolute paths so that the user doesn't have to worry about the
environment they call it with.

### CODE REVIEW

File: pkgs/build-support/build-fhs-userenv/default.nix [hunk 1]
```diff
@@ -1,8 +1,9 @@
-{ runCommand, lib, writeText, writeScriptBin, stdenv, ruby } :
-{ env, runScript ? "bash", extraBindMounts ? [], extraInstallCommands ? "" } :
+{ runCommand, lib, writeText, writeScriptBin, stdenv, bash, ruby } :
+{ env, runScript ? "${bash}/bin/bash", extraBindMounts ? [], extraInstallCommands ? "" } :
```
Comment (kevincox) [inline pkgs/build-support/build-fhs-userenv/default.nix]: This one could be left and that way the user would get whatever bash is in their path as the default which could be considered a feature. I think it is best like this as it won't change depending on how you call it but either way makes sense.

**Comment by abbradar** [comment]
I have no problem regarding the change but how have you encountered a problem?  `bash` should be called inside FHS sandbox and we have `bashInteractive` there as a base package (see https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/build-fhs-chrootenv/env.nix#L50).

**Comment by kevincox** [comment]
I don't know what caused it but this issue came up a couple of days ago on master. [This commit](https://github.com/NixOS/nixpkgs/commit/9b33ec1764c34d5e9d542f59ce932209941b64f1) looks suspicious for triggering the issue but a good idea to do. Also you have linked `fhs-chrootenv` where as I was using `fhs-userenv`.

**Comment by kevincox** [comment]
The use case was calling this from a systemd service. (I'm running a fuzzer in an environment because the shell script doesn't play nicely with Nix). I suppose it rears it's head in this scenario because the path only has the bare minimum in this case, where as most other times you have bash in your path.

**Comment by abbradar** [comment]
This part of `build-fhs-chrootenv` is common -- it builds an actual FHS environment. Anyway, let's merge this, assumptions like that are invalid anyway (what if we'd change the default shell in FHS in future?).

**Pull request merged**